src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -240,28 +242,63 @@ void Add(INamedTypeSymbol? symbol)
                 }
             }
 
-            Add(compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemWebServicesWebMethodAttribute));
+            // Web attributes
+            Add(wellKnownTypeProvider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemWebServicesWebMethodAttribute));
```

Comment:
I could also check that the enclosing type is a controller.

Reply:
I think it is fine to not check enclosing type for the attribute case.